### DISCUSSION TITLE
Index on char(N) columns not getting picked up by SELECT queries

### DISCUSSION THREAD

**Discussion by cakir-enes** [go-jet/jet#415] [General]
Generated models for `char(N)` columns use `string` type so to query you write
`table.Account.SELECT(table.Account.AllColumns).WHERE(table.Account.L1Address.EQ(String("somthing")).Query..` but this wont use indexes. You have to actually write `CAST(String("somthing")).AS_CHAR(42)` which is pretty unintuitive imo `go-jet` should at least generate helper funcs and warn about this in docs.

```
CREATE TABLE "account" (
    "account_index" int8 PRIMARY KEY,
    "l1_address" char(42)
);
CREATE INDEX account_l1_address_ix_1 ON account (l1_address);
-- this wont help either
CREATE INDEX account_l1_address_ix ON account (l1_address, (l1_address::text));
```

**Comment by go-jet** [comment]
Yeah, I agree. Check my [comment](https://github.com/go-jet/jet/issues/273#issuecomment-2432099294) on related issue.

**Comment by go-jet** [comment]
In release v2.12.0, added new constructors for Postgres-specific character types, including `Text`, `Char`, and `VarChar`.
